modules/jdbc/src/integrationTest/java/org/apache/ignite/jdbc/ItJdbcMultiStatementSelfTest.java

```diff
@@ -52,48 +56,321 @@ public void setupTables() throws Exception {
+        resultSet = stmt.getResultSet();
+        assertNull(resultSet);
+        assertEquals(0, stmt.getUpdateCount());
+
+        // after commit
+        assertFalse(stmt.getMoreResults());
+        assertEquals(-1, stmt.getUpdateCount());
+    }
+
+    @Test
+    public void testTransactionQueryInsideOutside() throws Exception {
+        stmt.execute("START TRANSACTION; SELECT 1; COMMIT; SELECT 2;");
+        ResultSet resultSet = stmt.getResultSet();
+        assertNull(resultSet);
+        assertEquals(0, stmt.getUpdateCount());
+
+        // SELECT 1;
+        assertTrue(stmt.getMoreResults());
+        resultSet = stmt.getResultSet();
+        assertNotNull(resultSet);
+
+        // COMMIT;
+        assertFalse(stmt.getMoreResults());
+        resultSet = stmt.getResultSet();
+        assertNull(resultSet);
+        assertEquals(0, stmt.getUpdateCount());
+
+        // SELECT 2;
+        assertTrue(stmt.getMoreResults());
+        resultSet = stmt.getResultSet();
+        assertNotNull(resultSet);
+        assertEquals(-1, stmt.getUpdateCount());
+
+        // after
+        assertFalse(stmt.getMoreResults());
+        assertEquals(-1, stmt.getUpdateCount());
     }
 
-    /**
-     * Assert that script containing both h2 and non h2 (native) sql statements is handled correctly.
-     */
     @Test
-    public void testMixedCommands() throws Exception {
-        execute("CREATE TABLE public.transactions (pk INT, id INT, k VARCHAR, v VARCHAR, PRIMARY KEY (pk, id)); "
-                + "CREATE INDEX transactions_id_k_v ON public.transactions (id, k, v) INLINE_SIZE 150; "
-                + "INSERT INTO public.transactions VALUES (1,2,'some', 'word') ; "
-                + "CREATE INDEX transactions_k_v_id ON public.transactions (k, v, id) INLINE_SIZE 150; "
-                + "CREATE INDEX transactions_pk_id ON public.transactions (pk, id) INLINE_SIZE 20;");
+    public void testDmlInsideTransaction() throws Exception {
+        stmt.execute("START TRANSACTION; INSERT INTO TEST_TX VALUES (5, 19, 'Nick'); COMMIT");
+        assertEquals(-1, getResultSetSize());
+    }
+
+    @Test
+    public void testAutoCommitFalse() throws Exception {
+        conn.setAutoCommit(false);
+        assertThrows(SQLException.class, () -> stmt.execute("COMMIT"));
+
+        boolean res = stmt.execute("SELECT 1;COMMIT");
+        assertTrue(res);
+        assertNotNull(stmt.getResultSet());
+        assertThrows(SQLException.class, () -> stmt.getMoreResults());
+    }
+
+    @Test
+    public void testPreviousResultSetIsClosedExecute() throws Exception {
+        boolean res = stmt.execute("SELECT ID FROM TEST_TX; SELECT 1;");
+        assertEquals(true, res);
+        stmt.getResultSet();
+        ResultSet rs = stmt.getResultSet();
+
+        res = stmt.getMoreResults();
+        assertEquals(true, res);
+
+        assertTrue(rs.isClosed());
+        assertEquals(1, getResultSetSize());
+
+        stmt.execute("SELECT 1; SELECT 2; SELECT 3;");
```

Comment:
I see that the following test case hangs
```
        stmt.execute("START TRANSACTION; SELECT * FROM TABLE(system_range(0, 2000)); COMMIT;");
        stmt.getMoreResults();
        stmt.getMoreResults();
```
and the following finished with error
```
        stmt.execute("START TRANSACTION; SELECT * FROM TABLE(system_range(0, 2000)); COMMIT;");
        stmt.getMoreResults();
        stmt.getResultSet().close();
        stmt.getMoreResults(); //  Failed to find resource with id: 6
```
Can you investigate this issue?

Reply:
tx scenario is fixed, appropriate test appended.